File: packages/api-client/src/domains/address.ts
/**
 * 地址域 API
 */

import { post } from '../client';
import type { UserAddress } from '@fe/shared';

export const address = {
  async list(): Promise<UserAddress[]> {
    return post<UserAddress[]>('/v1/user/address/list');
  },

  async create(input: {
    label?: string;
    recipient: string;
    phone: string;
    province: string;
    city: string;
    district: string;
    address: string;
    postalCode?: string;
    isDefault?: boolean;
  }): Promise<UserAddress> {
    return post<UserAddress>('/v1/user/address/create', input);
  },

  async update(input: {
    id: string;
    label?: string;
    recipient?: string;
    phone?: string;
    province?: string;
    city?: string;
    district?: string;
    address?: string;
    postalCode?: string;
    isDefault?: boolean;
  }): Promise<UserAddress> {
    return post<UserAddress>('/v1/user/address/update', input);
  },

  async remove(id: string): Promise<null> {
    return post<null>('/v1/user/address/delete', { id });
  },
};

File: apps/android/app/src/main/java/com/example/shop/feature/user/data/model/Address.kt
package com.example.shop.feature.user.data.model

import kotlinx.serialization.Serializable

@Serializable
data class Address(
    val id: String,
    val userId: String = "",
    val label: String? = null,
    val recipient: String,
    val phone: String,
    val province: String,
    val city: String,
    val district: String,
    val address: String,
    val postalCode: String? = null,
    val isDefault: Boolean = false,
    val createdAt: String = "",
    val updatedAt: String = "",
)

@Serializable
data class AddressCreateRequest(
    val label: String? = null,
    val recipient: String,
    val phone: String,
    val province: String,
    val city: String,
    val district: String,
    val address: String,
    val postalCode: String? = null,
    val isDefault: Boolean = false,
)

@Serializable
data class AddressUpdateRequest(
    val id: String,
    val label: String? = null,
    val recipient: String? = null,
    val phone: String? = null,
    val province: String? = null,
    val city: String? = null,
    val district: String? = null,
    val address: String? = null,
    val postalCode: String? = null,
    val isDefault: Boolean? = null,
)

@Serializable
data class AddressDeleteRequest(val id: String)
